homeassistant/components/tuneblade_remote/config_flow.py

```diff
@@ -0,0 +1,130 @@
+"""Adds config flow for TuneBlade Remote."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from pytuneblade import TuneBladeApiClient
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.service_info.zeroconf import ZeroconfServiceInfo
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TuneBladeConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Config flow for TuneBlade Remote."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize."""
+        self._discovery_info: dict[str, Any] = {}
+        self._title_placeholders: dict[str, str] = {}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Manual user setup step."""
+        errors = {}
+
+        if user_input is not None:
+            name = user_input["name"].strip()
+            host = user_input["host"]
+            port = user_input["port"]
+
+            session = async_get_clientsession(self.hass)
+            client = TuneBladeApiClient(host, port, session)
+            try:
+                devices = await client.async_get_data()
+            except Exception:
+                _LOGGER.exception("Failed to connect to TuneBlade")
+                errors["base"] = "cannot_connect"
+            else:
+                if not devices:
+                    errors["base"] = "cannot_connect"
+
+            if not errors:
+                unique_id = f"{name}_{host}_{port}"
+                await self.async_set_unique_id(unique_id)
```

Comment:
we should pick a better unique id, or we should match using `self._async_abort_entries_match`